Lib/fontTools/feaLib/builder.py

```diff
@@ -259,12 +259,16 @@ def add_lookup_to_feature_(self, lookup, feature_name):
             key = (script, lang, feature_name)
             self.features_.setdefault(key, []).append(lookup)
 
-    def get_lookup_(self, location, builder_class):
+    def get_lookup_(self, location, builder_class, mapping=None):
         if (
             self.cur_lookup_
             and type(self.cur_lookup_) == builder_class
             and self.cur_lookup_.lookupflag == self.lookupflag_
             and self.cur_lookup_.markFilterSet == self.lookupflag_markFilterSet_
+            and (
+                builder_class != AnySubstBuilder
+                or self.cur_lookup_.can_add_mapping(mapping)
+            )
```

Comment:
I’d add `can_add_mapping` to base `LookupBuilder` class and make it always return true, to avoid the `builder_class != AnySubstBuilder` check.